archive/new_studyinsweden.py

```diff
@@ -0,0 +1,225 @@
+            #! Extract web elmements
+            soup = BeautifulSoup(response.text, 'html.parser')
+            paragraphs = soup.find_all(['p', 'h1', 'h2', 'h3'])
+            text = " ".join([p.get_text().strip() for p in paragraphs])
+            text_start = text[:1000]  # take first 1000 characters to assess the language
+
+            # Check if the website has supported languages
+            detected = self.translator.detect(text_start)
+            if detected.lang not in ['sv', 'en', 'vi']:
+                # print(f'Page at {url} is written in an unsupported language: {detected.lang}')
+                return
+
+            date_tag = soup.find('p', class_='ahjalpfunktioner')
+            if date_tag:
+                time_tag = date_tag.find('time')
+                date = time_tag.get_text() if time_tag else datetime.datetime.now().strftime("%Y-%m-%d")
+            else:
+                date = datetime.datetime.now().strftime("%Y-%m-%d")
+
+            title = soup.title.text.strip()
+            title = self.translate_text(title)
+            chunks = self.translate_text(text)
+
+            entries = {}
+            for idx, chunk in enumerate(chunks):
+                entries[str(idx)] = {
+                    "chunk-id": str(idx),
+                    "source": url,
+                    "title": title[0],
+                    "chunk": chunk,
+                    "updated": date,
+                }
+            for key in entries:
+                self.data_buffer.append(entries[key])
+            self.write_to_file(output_file)
+
+            #! Recursion
+            # Find all links on the page
+            links = soup.find_all('a')
+            # Recursively crawl each of the links found on the page
+            for link in links:
+                href = link.get('href')
+                if (href == None) or (href == '/') or (len(href)==0) or ('#' in href): continue
+                if href and href.startswith('https'):
+                    new_url = href
+                else:
+                    new_url = urljoin(url, href)
+                if (not new_url.startswith('https://studyinsweden.se/') \
+                    or (new_url.startswith('https://studyinsweden.se/blogs'))):
+                    continue
+                
+                if urlparse(new_url).netloc == urlparse(url).netloc and new_url not in self.visited_urls:
+                    time.sleep(0.1)
+                    self.crawl_website(new_url, output_file, depth=depth - 1)
+
+        #! Give error message when connection fails
+        except requests.exceptions.RequestException as err:
+            print(f"RequestException: {err}")
+        except requests.exceptions.HTTPError as errh:
+            print(f"HTTPError: {errh}")
+        except requests.exceptions.ConnectionError as errc:
+            print(f"ConnectionError: {errc}")
+        except Exception as e:
+            print(f"An error occurred while processing {url}: {str(e)}")
```

Comment:
Can replace this with 
```python
except (HTTPError, RequestException, ConnectionError) as err:
    print(f'Error when connecting: {err}')
```

Reply:
Wait for @ntluong95 and @tuananhdao to comment on this